### DISCUSSION TITLE
Has `.future` modifier gone away?

### DISCUSSION THREAD

**Discussion by tomgilder** [rrousselGit/riverpod#1933] [Q&A] [answered]
In RIverpod 1 we widely used `.future` with a `StateNotifier<AsyncValue<T>>`, like this:

```dart
final myProvider = StateNotifierProvider<MyProvider, AsyncValue<String>>(
  (ref) => MyProvider(),
);

class MyProvider extends StateNotifier<AsyncValue<String>> {
  MyProvider() : super(const AsyncValue.loading()) {
    // Load something
  }
}
```

```dart
await ref.watch(myProvider.future);
```

The `.future` modifier seems to have gone away in Riverpod 2, but I couldn't find anything in the changelog about it - has it gone for good and we need to move to `AsyncNotifier`?

Thanks!

**Comment by rrousselGit** [comment, marked as answer]
This wasn't a thing in 1.0 either.
It was only there during the -dev version of the 2.0

It was later removed as some scenarios could not be handled this way. Instead use the new AsyncNotifier
